sys/riscv/riscv/trap.c

```diff
@@ -424,8 +437,15 @@ do_trap_supervisor(struct trapframe *frame)
 		    (__cheri_addr unsigned long)frame->tf_sepc);
 		break;
 #if __has_feature(capabilities)
-	case EXCP_LOAD_CAP_PAGE_FAULT:
 	case EXCP_STORE_AMO_CAP_PAGE_FAULT:
+		/* This could be a cap-dirtying fast trap */
+		if (try_capdirty(kernel_map->pmap,
```

Comment:
Why is this not following the same path as normal dirty tracking through data_abort and pmap_fault_fixup? Yeah, sure, you probably want all the speed you can get, but doing your own ad-hoc thing isn't great for maintainability or readability.

Reply:
No particularly *good* reason, I suppose, save that `EXCP_STORE_AMO_CAP_PAGE_FAULT` was where it was prior to my getting there.  Probably both `CAP_PAGE_FAULTS` want to be folded in to `data_abort`?  If that happens I'll happily rebase atop it.